### ISSUE TITLE
Can't get options to work for timepicker

### ISSUE THREAD

**Issue by dcweems** [jonthornton/jquery-timepicker#455]
I'm missing something.  This should be easy.  I can't seem to get any of the options to work for timepicker.  Here's what I'm doing:
<html>
	<head>
		<meta content="text/html; charset=ISO-8859-1" http-equiv="content-type">
		<link rel="stylesheet" type="text/css" href="include/Styles.css">
		<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
  		<link rel="stylesheet" href="https://ajax.googleapis.com/ajax/libs/jqueryui/1.11.4/themes/redmond/jquery-ui.css">
		<script src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.11.4/jquery-ui.min.js"></script>
		<link rel="stylesheet" type="text/css" href="thirdParty/jquery.timepicker.css">
		<script src="thirdParty/jquery.timepicker.js"></script>
                <script type="text/javascript">
			$(document).ready(function() {
				$('#timeTest').timepicker({ 'scrollDefault': 'now' });
				$("#datePicker").datepicker();
			});
                </script>
	</head>

and later:

<p style="text-align: right;">	
      <label>Time of Service</label>
      <input type="text" class='time' id="timeTest" step='15'/>
</p>

As you see I've tried setting options in the ready function as well as with an html attribute to set the step.  I can't seem to find a variation that works.  But I do see the basic timepicker but I can't set the step to 15 minutes or bring it up on the current time.

What am I doing wrong?  Thanks

**Comment by jonthornton** [comment]
Can you post a jsFiddle or code sample that demonstrates the problem?

**Comment by dcweems** [comment]
Interesting.  I wrote the code in my comment but now I see it's gone.  Note the big empty space.  Do you have some sort of filter that removes it?

**Comment by jonthornton** [comment]
No, not sure what happened there.

**Comment by dcweems** [comment]
I'll try again this evening.  I'm at work now but the web work is something I do at home.

**Comment by dcweems** [comment]
Hey Jon,

Hopefully this goes through.  First the <head>:

        <meta content="text/html; charset=ISO-8859-1"
http-equiv="content-type">
        <title>Progress Note</title>
        <link rel="stylesheet" type="text/css" href="include/Styles.css">
        <script src="
https://ajax.googleapis.com/ajax/libs/jquery/1.11.1/jquery.min.js"></script>
          <link rel="stylesheet" href="
https://ajax.googleapis.com/ajax/libs/jqueryui/1.11.4/themes/redmond/jquery-ui.css
">
        <script src="
https://ajax.googleapis.com/ajax/libs/jqueryui/1.11.4/jquery-ui.min.js
"></script>
        <script src="thirdParty/jquery.validate.js"></script>
        <link rel="stylesheet" type="text/css"
href="thirdParty/jquery.timepicker.css">
        <script src="thirdParty/jquery.timepicker.js"></script>
       <script type="text/javascript">
            $(document).ready(function() {
                $('#timeTest').timepicker({ 'scrollDefault': 'now' });
                $("#datePicker").datepicker();
            });
       </script>

And  then in the <body>:

<input type="text" class='time' id="timeTest" step='15'/>

I've also tried putting the function call in the <script></script> section
beneath the <input> but to no avail.  "thirdParty" is the directory where I
put your files as well as others that I use.

I hope you see the code now and see what the heck I'm missing.


On Thu, Nov 5, 2015 at 9:08 AM, Jon Thornton <notifications@github.com>
wrote:

> No, not sure what happened there.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jonthornton/jquery-timepicker/issues/455#issuecomment-154104663>
> .
>



-- 
Dan Weems
29 Adenois Lujan Road
Santa Fe, NM 87506
dan.weems@gmail.com
505-455-3668 (Home)
505-603-4692 (Cell)
505-982-0088 (Work)
<http://www.linkedin.com/in/danweems>

**Comment by jonthornton** [comment]
The step should be set as a data attribute: `<input ... data-step="15" />`. I don't see anything obviously wrong with the rest of the code, with regards to jquery-timepicker.

**Comment by dcweems** [comment]
Sigh.  That doesn't work either.  And I tried these too:

$('#startTimeOfSession').timepicker({ 'scrollDefault': 'now' });
$('#startTimeOfSession').timepicker({ 'step': 15 });

And they don't work.  Can't figure out why when it works in general but not
with the options.  So strange.  And I use other JQuery packages that work
fine.  Maybe I should try an older file?


On Sun, Nov 8, 2015 at 6:40 PM, Jon Thornton <notifications@github.com>
wrote:

> The step should be set as a data attribute: <input ... data-step="15" />.
> I don't see anything obviously wrong with the rest of the code, with
> regards to jquery-timepicker.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jonthornton/jquery-timepicker/issues/455#issuecomment-154899656>
> .
>



-- 
Dan Weems
29 Adenois Lujan Road
Santa Fe, NM 87506
dan.weems@gmail.com
505-455-3668 (Home)
505-603-4692 (Cell)
505-982-0088 (Work)
<http://www.linkedin.com/in/danweems>

**Comment by dcweems** [comment]
Here's another interesting note.  If I select the input field but don't
select anything and click again, I get a different list with something like
the following:

sdfg
3:00 pm
1345
34576
524y
asdfas

The values are not always what you see above but all seem related.  Might
this tell us what is wrong?

I thought maybe another package was interfering with yours but commenting
them all out didn't have an effect.



On Mon, Nov 9, 2015 at 12:34 PM, Dan Weems <dan.weems@gmail.com> wrote:

> Sigh.  That doesn't work either.  And I tried these too:
>
> $('#startTimeOfSession').timepicker({ 'scrollDefault': 'now' });
> $('#startTimeOfSession').timepicker({ 'step': 15 });
>
> And they don't work.  Can't figure out why when it works in general but
> not with the options.  So strange.  And I use other JQuery packages that
> work fine.  Maybe I should try an older file?
>
>
> On Sun, Nov 8, 2015 at 6:40 PM, Jon Thornton <notifications@github.com>
> wrote:
>
>> The step should be set as a data attribute: <input ... data-step="15" />.
>> I don't see anything obviously wrong with the rest of the code, with
>> regards to jquery-timepicker.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/jonthornton/jquery-timepicker/issues/455#issuecomment-154899656>
>> .
>>
>
>
>
> --
> Dan Weems
> 29 Adenois Lujan Road
> Santa Fe, NM 87506
> dan.weems@gmail.com
> 505-455-3668 (Home)
> 505-603-4692 (Cell)
> 505-982-0088 (Work)
> <http://www.linkedin.com/in/danweems>
>



-- 
Dan Weems
29 Adenois Lujan Road
Santa Fe, NM 87506
dan.weems@gmail.com
505-455-3668 (Home)
505-603-4692 (Cell)
505-982-0088 (Work)
<http://www.linkedin.com/in/danweems>

**Comment by dcweems** [comment]
OK, I fixed it!  What I did was downloaded your code (both js and css)
directly from your web page of the demos and replaced the ones I got from
downloading the zip file from GitHub.  No idea how they differ.  I'll leave
that to you.

Now one question.  I see you can do date/time start/end but I only want to
do the time start/end into two fields.  Can I do that?  Of course the idea
is to specify the start time and then the end time must be greater than the
start time.  If you can show me how to do that, awesome!

On Mon, Nov 9, 2015 at 12:45 PM, Dan Weems <dan.weems@gmail.com> wrote:

> Here's another interesting note.  If I select the input field but don't
> select anything and click again, I get a different list with something like
> the following:
>
> sdfg
> 3:00 pm
> 1345
> 34576
> 524y
> asdfas
>
> The values are not always what you see above but all seem related.  Might
> this tell us what is wrong?
>
> I thought maybe another package was interfering with yours but commenting
> them all out didn't have an effect.
>
>
>
> On Mon, Nov 9, 2015 at 12:34 PM, Dan Weems <dan.weems@gmail.com> wrote:
>
>> Sigh.  That doesn't work either.  And I tried these too:
>>
>> $('#startTimeOfSession').timepicker({ 'scrollDefault': 'now' });
>> $('#startTimeOfSession').timepicker({ 'step': 15 });
>>
>> And they don't work.  Can't figure out why when it works in general but
>> not with the options.  So strange.  And I use other JQuery packages that
>> work fine.  Maybe I should try an older file?
>>
>>
>> On Sun, Nov 8, 2015 at 6:40 PM, Jon Thornton <notifications@github.com>
>> wrote:
>>
>>> The step should be set as a data attribute: <input ... data-step="15" />.
>>> I don't see anything obviously wrong with the rest of the code, with
>>> regards to jquery-timepicker.
>>>
>>> —
>>> Reply to this email directly or view it on GitHub
>>> <https://github.com/jonthornton/jquery-timepicker/issues/455#issuecomment-154899656>
>>> .
>>>
>>
>>
>>
>> --
>> Dan Weems
>> 29 Adenois Lujan Road
>> Santa Fe, NM 87506
>> dan.weems@gmail.com
>> 505-455-3668 (Home)
>> 505-603-4692 (Cell)
>> 505-982-0088 (Work)
>> <http://www.linkedin.com/in/danweems>
>>
>
>
>
> --
> Dan Weems
> 29 Adenois Lujan Road
> Santa Fe, NM 87506
> dan.weems@gmail.com
> 505-455-3668 (Home)
> 505-603-4692 (Cell)
> 505-982-0088 (Work)
> <http://www.linkedin.com/in/danweems>
>



-- 
Dan Weems
29 Adenois Lujan Road
Santa Fe, NM 87506
dan.weems@gmail.com
505-455-3668 (Home)
505-603-4692 (Cell)
505-982-0088 (Work)
<http://www.linkedin.com/in/danweems>

**Comment by jonthornton** [comment]
Glad you got it working!

You're looking for [Datepair.js](https://github.com/jonthornton/Datepair.js).
